pigeon/base_msg.py

```diff
@@ -23,3 +23,31 @@ def deserialize(cls, data: str):
             An instantiation of the model using the JSON data.
         """
         return cls.model_validate_json(data)
+
+
+class AnnounceConnection(BaseMessage):
+    name: str
+    process_name: str
+    pid: int
+    hostname: str
+    connected: bool
+
+
+class RequestState(BaseMessage):
+    ...
+
+
+class UpdateState(BaseMessage):
+    name: str
+    process_name: str
+    pid: int
+    hostname: str
+    subscribed_to: list[str]
+
+msg_version = "1.0.0"
```

Comment:
I have been using package version since it must be incremented before the next version of the package can be pushed to PyPI. I am worried that a change could be made to the message definitions, but `msg_version` could be forgotten. However, this would break all backward compatibility between different versions of the Pigeon client. We probably need to discuss this more.

Reply:
#12 is one way to fix this.

Reply:
One issue is that if the message version is the same as package version, then multi message versions are in reality identical and in some universe where you're maintaining this at scale, you'd actually need to maintain the compatibility history.

Reply:
Let's leave it as is for now. If we decide to merge #13, we can then remove the hard-coded version here.